pkg/cloud/services/ec2/instances.go

```diff
@@ -653,7 +654,12 @@ func (s *Service) runInstance(role string, i *infrav1.Instance) (*infrav1.Instan
 
 	if i.Tenancy != "" {
 		input.Placement = &ec2.Placement{
-			Tenancy: &i.Tenancy,
+			Tenancy: ptr.To(string(i.Tenancy)),
+		}
+
+		if i.Tenancy == infrav1.TenancyHost && i.DedicatedHostPlacement != nil {
+			input.Placement.Affinity = i.DedicatedHostPlacement.Affinity
```

Comment:
These fields are labeled as optional; should we double check that they aren't nil as well? Or perhaps add a webhook to validate them before we get to this stage.

Reply:
the AWS sdk accepts nil as input, so I don't think we need to validate them.